src/http_client/mod.rs

```diff
@@ -0,0 +1,73 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use reqwest::Url;
+
+pub struct HttpClientImpl {}
+
+#[derive(serde::Deserialize, serde::Serialize, Debug)]
+pub struct CompilerMeetings {
+    items: Vec<CompilerMeeting>,
+}
+
+#[derive(serde::Deserialize, serde::Serialize, Debug)]
+pub struct Start {
+    #[serde(rename(deserialize = "dateTime"))]
+    date_time: String,
+}
+
+#[derive(serde::Deserialize, serde::Serialize, Debug)]
+pub struct CompilerMeeting {
+    summary: String,
+    #[serde(rename(deserialize = "htmlLink"))]
+    html_link: String,
+    #[serde(rename(deserialize = "originalStartTime"))]
+    original_start: Option<Start>,
+    start: Option<Start>,
+}
+
+#[async_trait]
+pub trait HttpClient {
+    async fn get_meetings(
+        &self,
+        start_date: String,
+        end_date: String,
+    ) -> Result<Vec<CompilerMeeting>>
+    where
+        Self: Sized;
+}
+
+#[async_trait]
+impl HttpClient for HttpClientImpl {
+    /// Retrieve all meetings from the Rust Compiler Team Calendar in a date range
+    // Google calendar API documentation:
+    // https://developers.google.com/calendar/api/v3/reference/events/list
+    // The API token needs only one permission: https://www.googleapis.com/auth/calendar.events.readonly
+    async fn get_meetings(
+        &self,
```

Comment:
self doesn't seem to be needed?

Reply:
correct, it's not used but it's needed to compile.

unsure how to refactor the code to remove that :man_shrugging: